### PR TITLE
Net5

### LINKED ISSUE #5737: Pull Sonarr commit 'New: Replace SmtpClient with Mailkit'
Pull https://github.com/Sonarr/Sonarr/commit/579c4433494a848aa75abb09d78af7e98ebb96a6

```
git cherry-pick -ex 579c4433494a848aa75abb09d78af7e98ebb96a6
```

This commit has 2 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

### LINKED ISSUE #5762: Pull Sonarr commit 'New: Require Encryption option for email'
Pull https://github.com/Sonarr/Sonarr/commit/e704ee617f379ff61fb1bf9fb4044d6b4eb12814

```
git cherry-pick -ex e704ee617f379ff61fb1bf9fb4044d6b4eb12814
```

This commit has 2 conflict(s)

You can ask me to `close all`, `open pr here` or `open pr everywhere`

### PR DESCRIPTION
**Pull request by ta264**
#### Database Migration
NO

#### Description
Build with net 5 (5.0.1)

#### Todos
- [x] 5.0.1
- [x] Revert to UserInteractive instead of the core workaround (fixed in 5)
- [x] Fix Restart due to dll reference instead of exe #5416
- [ ] Tests

#### Issues Fixed or Closed by this PR

* Fixes #5737
* Fixes #5762
* Closes #5764

### CODE REVIEW

**Comment by Qstick** [comment]
Updated this one a bit, should be good to go... Its being tested by a few users on Discord
- Changed from Preview 7 to Net5 5.0.1 final
- Bumped mono base to net472 from net462
- Fixed up UserInteractive code that caused issues in 3.1 with Windows service detection
- Switched to MailKit on recommendation from MSFT, SMTPClient is Obsolete
- Fixed reference to start up assembly (dll vs exe)
- Enabled built in Analyzers in lieu of FXCop due to deprecation

File: src/NzbDrone.Core/Notifications/Email/EmailService.cs [hunk 1]
```diff
@@ -24,27 +24,22 @@ public EmailService(Logger logger)
...
             {
-                credentials = new NetworkCredential(settings.Username, settings.Password);
-            }
+                Text = body
+            };
 
             try
             {
-                Send(email, settings.Server, settings.Port, settings.Ssl, credentials);
+                Send(email, settings.Server, settings.Port, settings.Username, settings.Password);
```
Comment (markus101) [inline src/NzbDrone.Core/Notifications/Email/EmailService.cs:42]: Thoughts on passing `settings` as one object?

File: src/NzbDrone.Core/Notifications/Email/EmailService.cs [hunk 2]
```diff
@@ -54,13 +49,20 @@ public void SendEmail(EmailSettings settings, string subject, string body, bool
             }
         }
 
-        private void Send(MailMessage email, string server, int port, bool ssl, NetworkCredential credentials)
+        private void Send(MimeMessage email, string server, int port, string user = null, string password = null)
```
Comment (markus101) [inline src/NzbDrone.Core/Notifications/Email/EmailService.cs:52]: Is there ever a case where `user` and `password` aren't explicitly passed in? If not they don't need default values.

**Review by markus101 — COMMENTED** [review]
Just a couple little things.

File: src/NzbDrone.Core/Notifications/Email/EmailService.cs (hunk 1 above)
Comment (Qstick) [reply src/NzbDrone.Core/Notifications/Email/EmailService.cs:42]: probably a good idea given we use almost everything

File: src/NzbDrone.Core/Notifications/Email/EmailService.cs (hunk 2 above)
Comment (Qstick) [reply src/NzbDrone.Core/Notifications/Email/EmailService.cs:52]: This goes away with the above change

**Pull request merged**